browse-next/src/components/DeviceSetupDefineMask.vue

```diff
@@ -60,21 +60,28 @@ const device = computed<ApiDeviceResponse | null>(() => {
 });
 
 const getExistingMaskRegions = async () => {
+  let regionsLength = 0;
+  const mostRecentTime = new Date(new Date().setDate(new Date().getDate() + 100));
   if (device.value) {
-    const existingMaskRegions = await getMaskRegionsForDevice(device.value.id);
-    console.log("Lenbth", existingMaskRegions.result.maskRegions.length);
-    for (let i = 0; i < existingMaskRegions.result.maskRegions.length; i++) {
-      points.value = existingMaskRegions.result.maskRegions[i].points;
+    const existingMaskRegions = await getMaskRegionsForDevice(
+      device.value.id,
+      mostRecentTime
+    );
+    // calculate number of regions present
+    while (existingMaskRegions.result[regionsLength]) {
+      regionsLength++;
+    }
+    for (let i = 0; i < regionsLength; i++) {
+      points.value = existingMaskRegions.result[i].points;
```

Comment:
Still unclear to me why we want to nest reactive values like this?  Why assign to points.value just to copy it to regionData in the next line, then points.value gets overwritten by the next iteration?